ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -279,6 +277,20 @@ applyHelper f cfg blk
   where
     globals = shelleyLedgerGlobals (blockConfigLedger cfg)
 
+    ledgerViewChanged :: SL.ShelleyState c -> SL.ShelleyState c -> Bool
+    ledgerViewChanged old new
+      | SL.nesEL old /= SL.nesEL new
+        -- When the epoch changed, we know the overlay schedule and the pool
+        -- distribution will have changed. This allows us to avoid checking
+        -- the equality of these huge data structures.
+      = True
+      | otherwise
+      = SL.lvProtParams oldLedgerView /= SL.lvProtParams newLedgerView ||
+        SL.lvGenDelegs  oldLedgerView /= SL.lvGenDelegs  newLedgerView
```

Comment:
@nc6 Do these two values ever change within an epoch or also only at the epoch boundary?

If the latter, then we can improve/simplify things even further.

Reply:
Protocol parameters only change at an epoch boundary. Genesis delegates may change within an epoch.

Reply:
Great, updated. Thanks!